app/src/main/java/com/skyyo/samples/features/healthConnect/HealthConnectViewModel.kt

```diff
@@ -31,22 +31,29 @@ class HealthConnectViewModel @Inject constructor(
     private val handle: SavedStateHandle,
     val application: Application,
 ) : ViewModel() {
-    private val healthConnectClient by lazy { HealthConnectClient.getOrCreate(application.applicationContext) }
+    val healthConnectClient = mutableStateOf<HealthConnectClient?>(null)
```

Comment:
not clear why this change was made. In this case surely `lazy` might be redundant, since we are showcasing this feature so 100% sure we are accessing it, but the approach is overall correct, and acts as a good reminder to delegate creation of objects which might not be accessed often

Reply:
@Skyyo, yup, but I needed the access to this object on the UI layer to obtain the permission contract right after the health connect feature availability is checked

Reply:
ok, clear. Let's use lateinit then, since we are in control over initialization. That will help to use less `.?`

Reply:
Previously this object wasn't needed for permission request. Though I'm not sure I've made a good modification cuz I'm using nullability checks now a lot. Maybe you can suggest a better approach in this case

Reply:
ok, clear. Let's then please use `lateinit` and remove observing it on UI layer ( since we don't need it to be observed). This fixes the `.?`

Reply:
But please ensure that we don't initialize it more than once, if composable `LaunchedEffect` is restarted (worst case scenario 0 simply check if `lateinit` variable is already initialized in the `initHealthSmth` function)